src/ProductConstructionService/ProductConstructionService.Api/Program.cs

```diff
@@ -45,12 +46,43 @@
 
 builder.Services.AddKustoClientProvider("Kusto");
 
+if (!builder.Environment.IsDevelopment())
+{
+    builder.AddAuthentication();
```

Comment:
I'm not sure what you mean by "talk to GitHub", we have a separate client that's using the token we provide. Everything in this method is about Authentication

Reply:
Oh, the `GitHubClientOptions` is not configured for all times we talk to GitHub?

Reply:
After looking into it, it appears it is. But there might be a small issue:
`GitHubClientOptions` are used by the `GitHubClientFactory`, which is registered in the same `dnceng-shared` extension method as the Authentication, and we can't call that one when in local development, since it is adding authentication.
So we got two options:
 - change the dnceng-shared code to register these in separate methods, and then change parts where they're consumed. There might be code in dnceng that uses this, so maybe it's not the best idea.
 - Just have an if and register the factory manually if in development mode

Reply:
or we can just register it twice, that doesn't hurt and the code will look nicer

Reply:
Registering twice is not a big deal most likely, yeah